src/components/ThemeProvider/ThemeProvider.js

```diff
@@ -0,0 +1,19 @@
+import React, { createContext, useState } from 'react';
+
+export const ThemeContext = createContext();
+
+const ThemeProvider = ({ children }) => {
+  const [isDarkTheme, setIsDarkTheme] = useState(false);
+
+  const toggleTheme = () => {
```

Comment:
Moglibyśmy tutaj dodać `useCallback`. Dzięki temu funkcja `toggleTheme` nie będzie się re-renderować przy każdej zmianie stanu
Tak by to mogło wyglądać:
`
const toggleTheme = useCallback(() => {
setIsDarkTheme((prev) => !prev);
}, [])
`

Reply:
> The useCallback hook is a built-in hook in React that lets you memoize a callback function by preventing it from being recreated on every render. In simple terms, it means that the callback function is cached and does not get redefined on every render.